diff --git a/cmd/sign.go b/cmd/sign.go
--- a/cmd/sign.go
+++ b/cmd/sign.go
@@ -1,8 +1,10 @@
 package cmd
 
 import (
+	"bufio"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"os"
 
 	"github.com/bnb-chain/tss/client"
 	"github.com/bnb-chain/tss/common"
@@ -36,5 +38,15 @@
 
 // TODO: use MessageBridge
 func setMessage() {
-	common.TssCfg.Message = "0"
+	if message := viper.GetString("message"); message != "" {
+		common.TssCfg.Message = message
+		return
+	}
+
+	reader := bufio.NewReader(os.Stdin)
+	message, err := common.GetString("please set message(in *big.Int.String() format) to be signed: ", reader)
+	if err != nil {
+		common.Panic(err)
+	}
+	common.TssCfg.Message = message
 }
